### PR TITLE
Gal 182/terraform set up

### PR DESCRIPTION
**Pull request by zainm97**
Ticket: https://nhsd-jira.digital.nhs.uk/browse/GAL-182

This PR creates an action in the pipeline to build the resources specified in the environment/dev folder. Another folder has been created under environment/prod which is intended to store the prod IaC when we get to writing it. 

The folder structure was heavily inspired but existing repos in the NHSDigital page. There are placeholder files such as data.tf, variables.tf, locals.tf, outputs.tf to name few that are included in the PR. These were committed to this PR as when the repo grows and the infrastructure gets more complex/involved, these files will be populated.

The current folder structure also contains a subdirectory called resources. This is to harbour the tf files that create the actual resources we need in AWS. The pipeline requires all terraform files to be in a single directory hence the need for the "Flatten Terraform Directory" step in the terraform action. 

Additional placeholder files such as iam.tf and lambda.tf have been included under the resources folder to emphasise the grouping structure we intend to maintain. 

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: .github/workflows/terraform.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        working-directory: ./terraform/environment/dev
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v3
+
+      - name: Configure AWS credentials from AWS account
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ vars.AWS_ROLE_ARN }}
+          aws-region: ${{ vars.AWS_REGION }}
```
Comment (msandhuNHS) [inline .github/workflows/terraform.yaml:26]: Won't need line 10 with this.

File: terraform/environment/dev/resources/s3.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+resource "aws_s3_bucket" "basic-s3-bucket" {
+  bucket        = "basic-s3-bucket"
+  force_destroy = true
+
+  tags = {
+    Name        = "My bucket"
```
Comment (msandhuNHS) [inline terraform/environment/dev/resources/s3.tf:6]: Maybe a more meaningful name for this bucket?

**Review by msandhuNHS — APPROVED** [review]
Some comments but not code breaking so approved.

**Review by LaithM96 — APPROVED** [review]
some comments for the code would be good to have for others to understand, but all checks looks fine

File: .github/workflows/terraform.yaml (hunk 1 above)
Comment (zainm97) [reply .github/workflows/terraform.yaml:26]: good spot, amended it to read in line 10

**Review by msandhuNHS — APPROVED** [review]

**Review by LaithM96 — APPROVED** [review]
all looks good :)

**Pull request merged**